src/commands/node.mjs

```diff
@@ -2314,8 +2352,829 @@ export class NodeCommand extends BaseCommand {
+              'serviceMap',
+              'stagingDir',
+              'stagingKeysDir',
+              'treasuryKey'
+            ])
+
+          config.curDate = new Date()
+          config.existingNodeIds = []
+
+          await self.initializeSetup(config, self.k8)
+
+          // set config in the context for later tasks to use
+          ctx.config = config
+
+          ctx.config.chartPath = await self.prepareChartPath(ctx.config.chartDirectory,
+            constants.FULLSTACK_TESTING_CHART, constants.FULLSTACK_DEPLOYMENT_CHART)
+
+          // initialize Node Client with existing network nodes prior to adding the new node which isn't functioning, yet
+          ctx.config.nodeClient = await this.accountManager.loadNodeClient(ctx.config.namespace)
+
+          const accountKeys = await this.accountManager.getAccountKeysFromSecret(FREEZE_ADMIN_ACCOUNT, config.namespace)
+          config.freezeAdminPrivateKey = accountKeys.privateKey
+
+          const treasuryAccount = await this.accountManager.getTreasuryAccountKeys(config.namespace)
+          const treasuryAccountPrivateKey = treasuryAccount.privateKey
+          config.treasuryKey = PrivateKey.fromStringED25519(treasuryAccountPrivateKey)
+
+          self.logger.debug('Initialized config', { config })
+        }
+      },
+      {
+        title: 'Identify existing network nodes',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeAddConfigClass} **/ ctx.config
+          config.serviceMap = await self.accountManager.getNodeServiceMap(
+            config.namespace)
+          for (/** @type {NetworkNodeServices} **/ const networkNodeServices of config.serviceMap.values()) {
+            config.existingNodeIds.push(networkNodeServices.nodeName)
+          }
+
+          return self.taskCheckNetworkNodePods(ctx, task, config.existingNodeIds)
+        }
+      },
+
+      {
+        title: 'Prepare gossip endpoints',
+        task: (ctx, task) => {
+          const config = /** @type {NodeDeleteConfigClass} **/ ctx.config
+          let endpoints = []
+          if (!config.gossipEndpoints) {
+            if (config.endpointType !== constants.ENDPOINT_TYPE_FQDN) {
+              throw new FullstackTestingError(`--gossip-endpoints must be set if --endpoint-type is: ${constants.ENDPOINT_TYPE_IP}`)
+            }
+
+            endpoints = [
+              `${Templates.renderFullyQualifiedNetworkPodName(config.namespace, config.nodeId)}:${constants.HEDERA_NODE_INTERNAL_GOSSIP_PORT}`,
+              `${Templates.renderFullyQualifiedNetworkSvcName(config.namespace, config.nodeId)}:${constants.HEDERA_NODE_EXTERNAL_GOSSIP_PORT}`
+            ]
+          } else {
+            endpoints = helpers.splitFlagInput(config.gossipEndpoints)
+          }
+
+          ctx.gossipEndpoints = this.prepareEndpoints(config.endpointType, endpoints, constants.HEDERA_NODE_INTERNAL_GOSSIP_PORT)
+        }
+      },
+      {
+        title: 'Prepare grpc service endpoints',
```

Comment:
I don't think this is needed for node delete

Reply:
Updated